### PR TITLE
Refactor analytics

### PR DESCRIPTION
**Pull request by dcodeIO**
Refactor analytics to its own file and run it only if DNT isn't explicitly requested. Though the right thing to do, this doesn't solve any [technical issues](https://www.w3.org/TR/tracking-dnt/) on its own.

### CODE REVIEW

File: lib/ga.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+if (navigator.doNotTrack != "1") (function() {
+  var script = document.createElement("script");
+  document.head.appendChild(script);
+  script.onload = function() {
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
```
Comment (yurydelendik) [inline lib/ga.js]: `gtag` needs to exposed on the global/window object. Use `window.gtag = ...`

File: lib/ga.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+if (navigator.doNotTrack != "1") (function() {
```
Comment (yurydelendik) [inline lib/ga.js]: Add a license header.

**Comment by dcodeIO** [comment]
> Please also squash commits.

GitHub also has a Squash-merge ability when selecting the merge action from the merge-button drop down. This results in those fancy "dcodeIO committed with yurydeleendik" commits. Can still squash if that's not an option, just let me know :)

**Comment by yurydelendik** [comment]
> selecting the squash action from the merge-button drop down

this functionality produces commit messages such as "Remove left-over alert" or "Address review comments" which do not make any sense. I prefer to have proper commit comment.

**Comment by yurydelendik** [comment]
Thank you for the patch.

**Comment by dcodeIO** [comment]
> this functionality produces commit messages such as "Remove left-over alert" or "Address review comments" which do not make any sense. I prefer to have proper commit comment.

That's correct, but it still can be edited before finally comitting it. I usually find this useful when merging contributions on my projects because it's just removing a line or two for me, while it is dealing with command line git for the contributor.

**Comment by yurydelendik** [comment]
> while it is dealing with command line git for the contributor

I understand you are not a novice git user :) But for novice git users, I would recommend to use `git commit --amend` for touch up commits (there is also an option in `git gui`). If I see a contributor struggling or ask for it, I'll merge it with squash option.

**Comment by dcodeIO** [comment]
I actually consider myself a novice Git user :) and I must admit that I've never used `--amend`, thanks for the hint.

**Comment by mbebenita** [comment]
I've been stuck at the novice Git user level for 7 years. I usually just send @yurydelendik some garbled up message about what I want to do, and he sends back a command.

**Pull request merged**